### ISSUE TITLE
✅ 🐞 Bug Report: Multiple Table of Contents create duplicate and mutliple anchors

### ISSUE THREAD

**Issue by mareklexuan** [elementor/elementor#11106]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

When using two Table of Contents on the page, each of them generates own set of anchors with identical IDs. They don't create one continuous set of anchors for a whole page. 

The second Table of Contents is then not able to reference the second set of anchors and they have the same IDs, so it is referencing different anchors, even to a different level of headings.

In the scenario below, Table of Contents 2 is set to H3, but will always link to the anchors for headings H2, as those anchors come first.

Scenario 1:
```

Table Of Contents 1 (H2)

- anchor 0 (id="elementor-toc__heading-anchor-0")
-- H2

- anchor 1 (id="elementor-toc__heading-anchor-1")
-- H2

- anchor 2 (id="elementor-toc__heading-anchor-2")
-- H2

Table Of Contents 2 (H3)

- anchor 0 (id="elementor-toc__heading-anchor-0")
-- H3

- anchor 1 (id="elementor-toc__heading-anchor-1")
-- H3

- anchor 2 (id="elementor-toc__heading-anchor-2")
-- H3
```

In addition, if I have two Table of Contents with overlapping targeted levels of headings, they create multiple anchors for the overlapping heading level.

```
Table Of Contents 1 (H2,H3)

- anchor 0 (id="elementor-toc__heading-anchor-0")
-- H2

- anchor 1 (id="elementor-toc__heading-anchor-1")
-- H2

- anchor 2 (id="elementor-toc__heading-anchor-2")
-- H2

Table Of Contents 2 (H3)

- anchor 3 (id="elementor-toc__heading-anchor-3")
- anchor 0 (id="elementor-toc__heading-anchor-0")
-- H3

- anchor 4 (id="elementor-toc__heading-anchor-4")
- anchor 1 (id="elementor-toc__heading-anchor-1")
-- H3

- anchor 5 (id="elementor-toc__heading-anchor-5")
- anchor 2 (id="elementor-toc__heading-anchor-2")
-- H3
```

I would assume that there would be just one set of anchors and the second table of contents would not create a new set of anchors but:

1/ continue with the numbering of anchors
2/ use the set created by the first anchor.

## Steps to reproduce

General page editing in Elementor, no special steps.

Table of Contents 1 is set to H2 (H2+H3 respectively)
Table of Contents 2 is set to H3

## Isolating the problem

- [x ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ x] This bug happens with a default WordPress theme active.
- [ x] I can reproduce this bug consistently using the steps above.

## Environment

System Info > [https://pastebin.com/mP8fZj5t](https://pastebin.com/mP8fZj5t
)

**Comment by jpminor** [comment]
Is there a solution to this?

**Comment by joaquin-alvarez** [comment]
Any workaround?

**Comment by dwhoop55** [comment]
Happens still with most up-to-date version with Elementor Pro. While you can tell the toc to only search a specific container, the generated toc may be visually correct, but all generated anchors near the headings won't reflect that they are in a container. They will be id'ed as all the same on the same level.

**Comment by faniabdo99** [comment]
Thank you Elementor for being the worst solution ever.
I am facing the same issue now and as @dwhoop55 said the whole scoping concept is not working, so you can basically have only one TOC per page.

**Comment by marowi85** [comment]
I am facing the same issue.

I think this could be solved if Elementor would use the headlines as anchors instead of the "elementor-toc__heading-anchor-0" with increasing numbers.

For example id="elementor-toc__lorem-ipsum-headline" for <h2>Lorem ipsum headline</h2> etc.

**Comment by beantherambo** [comment]
> I am facing the same issue.
> 
> @rami-elementor
> 
> I think this could be solved if Elementor would use the headlines as anchors instead of the "elementor-toc__heading-anchor-0" with increasing numbers.
> 
> For example id="elementor-toc__lorem-ipsum-headline" for < h2 >Lorem ipsum headline< /h2 > etc.

I have found out that this problem can be solved by adding "css id" to each Headling widget, the Table of contents menu  anchors will be sync with them. 

Suggest this setting should be add to the doc, thanks!

**Comment by davethaddeus** [comment]
@beantherambo 
Where did you place the heading widgets? I'm having a nightmare with multiple TOC's in a large manual.

Layout
--column-- TOC - H2 level only
 --inner section--
  ---column--- Chapter 1 content
  ---column---TOC H2-H4
 --inner section--
  ---column---Chapter 2 content
  ---column---TOC H2-H4
etc...

The left side is outside all inner section widgets and works perfectly.
Chapter 1 - TOC also works perfectly
Chapter 2 onwards - TOC display correctly, but all link to the original anchors.

Any help would be very much appreciated. Thanks.

**Comment by beantherambo** [comment]
> @beantherambo Where did you place the heading widgets? I'm having a nightmare with multiple TOC's in a large manual.
> 
> Layout --column-- TOC - H2 level only --inner section-- ---column--- Chapter 1 content ---column---TOC H2-H4 --inner section-- ---column---Chapter 2 content ---column---TOC H2-H4 etc...
> 
> The left side is outside all inner section widgets and works perfectly. Chapter 1 - TOC also works perfectly Chapter 2 onwards - TOC display correctly, but all link to the original anchors.
> 
> Any help would be very much appreciated. Thanks.

If your TOC displayed in correct ording you only have to check are those H2-H4 headline widgets have unique "CSS ID".

**Comment by davethaddeus** [comment]
> > @beantherambo Where did you place the heading widgets? I'm having a nightmare with multiple TOC's in a large manual.
> > Layout --column-- TOC - H2 level only --inner section-- ---column--- Chapter 1 content ---column---TOC H2-H4 --inner section-- ---column---Chapter 2 content ---column---TOC H2-H4 etc...
> > The left side is outside all inner section widgets and works perfectly. Chapter 1 - TOC also works perfectly Chapter 2 onwards - TOC display correctly, but all link to the original anchors.
> > Any help would be very much appreciated. Thanks.
> 
> @davethaddeus If your TOC links displayed in correct ording you only have to check are those H2-H4 headline widgets have unique "CSS ID" or not.

Thanks @beantherambo 
Works like a charm now.
All my headings we in the same text block. Breaking them out and giving each one an individual "css id" did the trick.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.15.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
